src/scheduler/JobScheduler.ts

```diff
@@ -11,66 +11,103 @@ import { getJobRepository } from '../repository/getJobRepository';
 import { ExecutionStatus, JobResult } from '../job/ExecutionInfo';
 import { DateTime } from 'luxon';
+import { jobDescriptionFromEntity, MomoJobDescription } from '../job/MomoJobDescription';
 
 export class JobScheduler {
   private jobHandle?: TimeoutHandle;
   private unexpectedErrorCount = 0;
+  private interval?: string;
 
-  constructor(private readonly job: Job, private readonly jobExecutor: JobExecutor, private readonly logger: Logger) {}
+  constructor(
+    private readonly jobName: string,
+    private readonly immediate: boolean,
+    private readonly jobExecutor: JobExecutor,
+    private readonly logger: Logger
+  ) {}
+
+  static forJob(job: Job, logger: Logger): JobScheduler {
+    const executor = new JobExecutor(job.handler, logger);
+    return new JobScheduler(job.name, job.immediate, executor, logger);
+  }
 
   public getUnexpectedErrorCount(): number {
     return this.unexpectedErrorCount;
   }
 
-  public async run(): Promise<JobScheduler> {
-    const interval = humanInterval(this.job.interval);
-    if (!interval || isNaN(interval)) {
-      throw MomoError.nonParsableInterval;
+  public isStarted(): boolean {
+    return this.jobHandle !== undefined;
+  }
+
+  public async getJobDescription(): Promise<MomoJobDescription | undefined> {
+    const jobEntity = await getJobRepository().findOne({ name: this.jobName });
+    if (!jobEntity) {
+      this.logger.error(
+        'get job description - job not found',
+        MomoErrorType.scheduleJob,
+        { name: this.jobName },
+        MomoError.jobNotFound
+      );
+      return;
     }
+    const schedulerStatus = this.interval !== undefined ? { interval: this.interval, started: true } : undefined;
+    return { ...jobDescriptionFromEntity(jobEntity), schedulerStatus };
+  }
+
+  public async start(): Promise<void> {
+    this.stop();
 
-    const [jobEntity] = await getJobRepository().find({ name: this.job.name });
+    const jobEntity = await getJobRepository().findOne({ name: this.jobName });
     if (!jobEntity) {
       this.logger.error(
         'cannot schedule job',
         MomoErrorType.scheduleJob,
-        { name: this.job.name },
+        { name: this.jobName },
         MomoError.jobNotFound
       );
-      return this;
+      return;
     }
-    const delay = calculateDelay(interval, this.job.immediate, jobEntity);
+
+    const interval = humanInterval(jobEntity.interval);
+    if (!interval || isNaN(interval)) {
+      throw MomoError.nonParsableInterval;
+    }
+
+    this.interval = jobEntity.interval;
+
+    const delay = calculateDelay(interval, this.immediate, jobEntity);
     this.jobHandle = setIntervalWithDelay(this.executeConcurrently.bind(this), interval, delay);
 
     this.logger.debug(`scheduled job to run at ${DateTime.now().plus({ milliseconds: delay }).toISO()}`, {
-      name: this.job.name,
+      name: this.jobName,
       interval,
       delay,
     });
-    return this;
   }
 
   public stop(): void {
```

Comment:
Do we have any chance of stopping currently running executions? I just wonder what happens if we have a long running job and we want to e.g. shut down or restart the schedule. What will happen with the execution? How do we make sure the running count in the db is correctly lowered?

Reply:
I would treat this separately - we have an open issue on what to do with running jobs when the scheduler is stopped

Reply:
Ah yeah... I forgot about #11
OK, let's keep that separately.